pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -477,6 +482,22 @@ func (cb *ClusterBuilder) buildInboundClusterForPortOrUDS(clusterPort int, bind
 				},
 			},
 		}
+		// add extra source addresses to cluster builder
```

Comment:
How does envoy pick which one to use?

Reply:
Hi @howardjohn, there were envoy changes to support this pick.  The basic doc is here: https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/address.proto#config-core-v3-bindconfig. And the issue [api, extend BindConfig to multiple addresses](https://github.com/envoyproxy/envoy/issues/9811) includes all related PRs, and the API changes PR for listener is: [PR#22639](https://github.com/envoyproxy/envoy/pull/22639) and [PR#22639](https://github.com/envoyproxy/envoy/pull/22639).

Reply:
It would be nice to comment in the go how it picks

Reply:
Hi @howardjohn, added the comments for this implementation, including the issue and PR.